core/src/cluster_slots_service/cluster_slots.rs

```diff
@@ -201,16 +356,15 @@ impl ClusterSlots {
         slot: Slot,
         repair_peers: &[ContactInfo],
     ) -> (Vec<u64>, Vec<usize>) {
-        let mut weights = Vec::with_capacity(repair_peers.len());
-        let mut indices = Vec::with_capacity(repair_peers.len());
-
         let Some(slot_peers) = self.lookup(slot) else {
```

Comment:
do not allocate before the check, this way we do not need to drop values immediately after allocation